### PR TITLE
NIAD-1647: DocumentReference: Practitioner author doesn't generate NarrativeStatement/participant

### PR DESCRIPTION
**Pull request by rorysproule1**

### CODE REVIEW

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java [hunk 1]
```diff
@@ -47,7 +51,8 @@ public String mapDocumentReferenceToNarrativeStatement(final DocumentReference d
         final NarrativeStatementTemplateParametersBuilder builder = NarrativeStatementTemplateParameters.builder()
             .narrativeStatementId(narrativeStatementId)
             .availabilityTime(getAvailabilityTime(documentReference))
-            .hasReference(true);
+            .hasReference(true)
```
Comment (bartek-sarul) [inline service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java:54]: Is it correct to set this to "true" if `buildParticipant(...)` can return an empty string [here](https://github.com/nhsconnect/integration-adaptor-gp2gp/pull/373/files#diff-aded27c693f2c511de5d4e319a801f6311120026c83bfb4b21a81bc5b620d77eR79) ?

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapperTest.java [hunk 2]
```diff
@@ -94,7 +94,8 @@ public void setUp() throws IOException {
         lenient().when(supportedContentTypes.isContentTypeSupported("application/octet-stream")).thenReturn(false);
         lenient().when(timestampService.now()).thenReturn(Instant.parse("2021-08-18T12:00:00.00Z"));
 
-        mapper = new DocumentReferenceToNarrativeStatementMapper(messageContext, supportedContentTypes, timestampService);
+        mapper = new DocumentReferenceToNarrativeStatementMapper(messageContext, supportedContentTypes, timestampService,
+                new ParticipantMapper());
```
Comment (bartek-sarul) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapperTest.java:98]:
```suggestion
        mapper = new DocumentReferenceToNarrativeStatementMapper(
            messageContext, supportedContentTypes, timestampService, new ParticipantMapper());
```

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapperComponentTest.java [hunk 3]
```diff
@@ -120,7 +121,8 @@ public void setUp() {
             participantMapper, multiStatementObservationHolderFactory);
         SpecimenMapper specimenMapper = new SpecimenMapper(messageContext, specimenObservationMapper, randomIdGeneratorService);
         DocumentReferenceToNarrativeStatementMapper documentReferenceToNarrativeStatementMapper
-            = new DocumentReferenceToNarrativeStatementMapper(messageContext, new SupportedContentTypes(), timestampService);
+            = new DocumentReferenceToNarrativeStatementMapper(messageContext, new SupportedContentTypes(), timestampService,
+                participantMapper);
```
Comment (bartek-sarul) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EhrExtractMapperComponentTest.java:125]:
```suggestion
            = new DocumentReferenceToNarrativeStatementMapper(
                messageContext, new SupportedContentTypes(), timestampService, participantMapper);
```

File: service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java [hunk 4]
```diff
@@ -105,7 +106,8 @@ public void setUp() {
         DiaryPlanStatementMapper diaryPlanStatementMapper
             = new DiaryPlanStatementMapper(messageContext, codeableConceptCdMapper, participantMapper);
         DocumentReferenceToNarrativeStatementMapper documentReferenceToNarrativeStatementMapper
-            = new DocumentReferenceToNarrativeStatementMapper(messageContext, new SupportedContentTypes(), timestampService);
+            = new DocumentReferenceToNarrativeStatementMapper(messageContext, new SupportedContentTypes(), timestampService,
+                participantMapper);
```
Comment (bartek-sarul) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapperTest.java:110]:
```suggestion
            = new DocumentReferenceToNarrativeStatementMapper(
                messageContext, new SupportedContentTypes(), timestampService, participantMapper);
```

File: service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java [hunk 5]
```diff
@@ -144,7 +144,8 @@ public void setUp() {
             multiStatementObservationHolderFactory);
         SpecimenMapper specimenMapper = new SpecimenMapper(messageContext, specimenObservationMapper, randomIdGeneratorService);
         DocumentReferenceToNarrativeStatementMapper documentReferenceToNarrativeStatementMapper
-            = new DocumentReferenceToNarrativeStatementMapper(messageContext, new SupportedContentTypes(), timestampService);
+            = new DocumentReferenceToNarrativeStatementMapper(messageContext, new SupportedContentTypes(), timestampService,
+                participantMapper);
```
Comment (bartek-sarul) [inline service/src/test/java/uk/nhs/adaptors/gp2gp/uat/EhrExtractUATTest.java:148]:
```suggestion
            = new DocumentReferenceToNarrativeStatementMapper(
                messageContext, new SupportedContentTypes(), timestampService, participantMapper);
```

**Review by bartek-sarul — CHANGES_REQUESTED** [review]

File: service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java (hunk 1 above)
Comment (bbell-kainos) [reply service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/DocumentReferenceToNarrativeStatementMapper.java:54]: I believe so as reference is do to with the external document, not the participant. See this mustache template for narrative statement: https://github.com/nhsconnect/integration-adaptor-gp2gp/blob/main/service/src/main/resources/templates/ehr_narrative_statement_template.mustache

**Pull request merged**